src/frame.js

```diff
@@ -0,0 +1,67 @@
+function Frame() {
+    this.STRIKE_SCORE = 10;
+    this.firstScore = null;
+    this.secondScore = null;
+    this.finalFrameScore = null;
+    this.strike = false;
+    this.spare = false;
+    this.finalised = false;
+}
+
+Frame.prototype.setFirstScore = function(score){
+    if (score == this.STRIKE_SCORE) {
+        this.strike = true;
+    }
+    this.firstScore = score;
+}
+
+Frame.prototype.setSecondScore = function(score){
+    if (this.isStrike()) {
+        return null;
+    } else {
```

Comment:
you don't need an else after a return.
```js
if (this.isStrike()) {
        return null;
if((score + this.getFirstScore()) === this.STRIKE_SCORE) {
    this.spare = true;
}
this.finalised = true;
this.secondScore = score;
this.setFinalFrameScore(this.getFirstScore() + this.getSecondScore());
```

Reply:
hmm interesting, I never knew that. thanks!